shard.go

```diff
@@ -16,7 +16,7 @@ type Metadata struct {
 }
 
 type cacheShard struct {
-	hashmap     map[uint64]uint32
+	hashmap     map[uint64]uint64
```

Comment:
does it belong to this  PR?

Reply:
no it doesn't, will revert it and wait for #377